test/suite/test_compact07.py

```diff
@@ -0,0 +1,212 @@
+        
+        # Check the background compaction server stats. We should have skipped at least once and 
+        # been successful at least once.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        skipped = stat_cursor[stat.conn.background_compact_skipped][2]
+        success = stat_cursor[stat.conn.background_compact_success][2]
+        self.assertGreater(skipped, 0)
+        self.assertGreater(success, 0)
+        stat_cursor.close()
+        
+        # Open a new session for foreground compaction, while leaving the background
+        # compaction server running.
+        session2 = self.conn.open_session()
+        # Use a free_space_target that is guaranteed to run on the small file.
+        session2.compact(uri_small,f'free_space_target={small_file_free_space - 1}MB')
+
+        time.sleep(1)
+        
+        # Periodically check that foreground compaction has done some work on the small table.
+        pages_rewritten = self.get_pages_rewritten(uri_small)
+        while (pages_rewritten == 0):
+            time.sleep(1)
+            pages_rewritten = self.get_pages_rewritten(uri_small)
+                    
+        # Disable the background compaction server.
+        self.session.compact(None,'background=false')
+        
+        # Check that the background server is no longer running.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 0)
+        stat_cursor.close()
+    
+    # Test the background server while creating and dropping tables.
+    def test_background_compact_table_ops(self):
+        # Trigger the background compaction server first, before creating tables.
+        self.session.compact(None,'background=true,free_space_target=1MB')
+
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+                
+        # Delete some data from the half the tables.
+        for i in range(self.n_tables):
+            if (i % 2 == 0):
+                self.delete_range(f'{self.uri_prefix}_{i}')
+                
+        # Periodically check that background compaction has done some work on the first table.
+        pages_rewritten = self.get_pages_rewritten(f'{self.uri_prefix}_0')
+        while (pages_rewritten == 0):
+            time.sleep(1)
+            pages_rewritten = self.get_pages_rewritten(f'{self.uri_prefix}_0')
+            
+        # Now drop all the tables.
+        for i in range(self.n_tables):
+            self.dropUntilSuccess(self.session, f'{self.uri_prefix}_{i}')
```

Comment:
This one is interesting as compaction may be blocking this call forever. That's something @marcbutler pointed out and the code does not handle this yet.

Reply:
Okay, I think that's an important point to test then. Should I comment this out as a fixme for now until it can handle that?

Reply:
Yes, it is a very interesting test. Please disable it and add a FIXME comment with WT-11450 which has been created to address this.